qubesappmenus/__init__.py

```diff
@@ -662,16 +663,26 @@ def retrieve_list(path):
 def main(args=None, app=None):
     """main function for qvm-appmenus tool"""
     args = parser.parse_args(args=args, app=app)
+    if not args.all_domains and not args.domains:
+        parser.error("one of the arguments --all VMNAME is required")
     appmenus = Appmenus()
     if args.source is not None:
         args.source = args.app.domains[args.source]
     if args.template is not None:
         args.template = args.app.domains[args.template]
-    for vm in args.domains:
+    if args.all_domains:
+        domains = args.app.domains
+    else:
+        domains = args.domains
+    for vm in domains:
+        if str(vm) == 'dom0':
+            continue
         # allow multiple actions
         # for remove still use just VM name (str), because VM may be already
-        # removed
+        # removed. It may be incompatible with --all option.
```

Comment:
I'm not sure if needed, but there is a way to get `--all` working even with removed VMs - for remove, iterate over `basedir` (as in - menu entries stored locally, regardless of VM being removed already or not).